src/SFA.DAS.LearnerDataMismatches.Web/Pages/Learner.cshtml.cs

```diff
@@ -4,54 +4,62 @@
+        public IEnumerable<Domain.CollectionPeriod> CurrentYearDataLocks { get; private set; } = Enumerable.Empty<Domain.CollectionPeriod> ();
+        public IEnumerable<Domain.CollectionPeriod> PreviousYearDataLocks { get; private set; } = Enumerable.Empty<Domain.CollectionPeriod> ();
 
+        public string LearnerName { get; private set; }
+        public string EmployerName { get; private set; }
+        public string EmployerId { get; private set; }
+        public string ProviderName { get; private set; }
+        public string ProviderId { get; private set; }
+        public bool HasDataLocks { get; private set; }
         public IEnumerable<string> DataLockNames =>
-            NewCollectionPeriods
+            CurrentYearDataLocks
             .SelectMany (x => x.DataLocks)
             .Select (x => x.ToString ())
             .OrderBy (x => x);
 
         public IEnumerable<DataLockHelpCentreLink> DataLockLinks =>
-            NewCollectionPeriods
-            .SelectMany (x => x.DataLocks)
-            .Distinct ()
-            .Select (DataLockHelpCentreLink.Create)
-            .OrderBy (x => x.Name);
-
-        public string EmployerName { get; set; }
-        public string EmployerId { get; set; }
-        public string ProviderName { get; set; }
-        public string ProviderId { get; set; }
+            CurrentYearDataLocks
+                .Concat(PreviousYearDataLocks)
+                .SelectMany (x => x.DataLocks)
+                .Distinct ()
+                .Select (DataLockHelpCentreLink.Create)
+                .OrderBy (x => x.Name);
+        
+        public AcademicYear AcademicYear => new AcademicYear(DateTime.Today);
         private readonly LearnerReportProvider learnerReportProvider;
 
         public LearnerModel (LearnerReportProvider learnerReportProvider) 
         {
             this.learnerReportProvider = learnerReportProvider;
         }
 
-        public async Task OnGetAsync () {
+        public async Task OnGetAsync() 
+        {
             if (!long.TryParse (Uln, out var uln))
                 throw new Exception ("Invalid ULN");
 
+            var report = await learnerReportProvider.BuildLearnerReport(uln, new [] {AcademicYear.Current, AcademicYear.Previous});
 
-            var report = await learnerReportProvider.BuildLearnerReport(uln);
-
+            HasDataLocks = report.HasDataLocks;
             LearnerName = report.Learner.Name;
-            NewCollectionPeriods = report.DataLocks;
             ProviderName = report.Provider.Name;
             ProviderId = report.Provider.Id;
             EmployerName = report.Employer.Name;
             EmployerId = report.Employer.Id;
+            if (report.HasDataLocks && report.DataLocks.ContainsKey(AcademicYear.Current)) CurrentYearDataLocks = report.DataLocks[AcademicYear.Current];
+            if (report.HasDataLocks && report.DataLocks.ContainsKey(AcademicYear.Previous)) PreviousYearDataLocks = report.DataLocks[AcademicYear.Previous];
```

Comment:
I think this could be made more readable, maybe:

```suggestion
            report.DataLocks.TryGetValue(AcademicYear.Previous, out PreviousYearDataLocks);
```

Reply:
Compiler does not allow using a Property as the out parameter. I will have to create a private variable for each property to use TryGetValue. I would like to keep the if condition so it is verbose. I will pull that out so it does not repeat.

Reply:
Yes, of course.  That's a shame.  

Is `HasDatalocks` redundant?  `ContainsKey` will always return false if there are no elements.

I'd be happy to keep two lines without an extra if block.

Reply:
HasDataLocks is used in the view to display the "no locks" message

Reply:
Sure, I wasn't suggesting removing the `HasDataLocks` property, just that checking it here is redundant :)

Reply:
The check is redundant, I kept it only to make a point and avoid checking dictionary.